### PR TITLE
Add a 'reverse' flag option to 'toctree'

### PR DESCRIPTION
**Pull request by JohnVillalovos**
I'm not sure the best way to do the unit tests. A little confused. So
help would be appreciated :)

Add the ability to reverse the list of entries when using 'toctree'.
This is likely most useful for people using the 'glob' flag option with
'toctree'

Change-Id: I1852479106c3d9c8253fd1625ced0077af0304e3

### CODE REVIEW

**Comment by tk0miya** [comment]
LGTM

File: doc/markup/toctree.rst [hunk 1]
```diff
@@ -123,6 +123,16 @@ tables of contents.  The ``toctree`` directive is the central element.
    toctree directive.  This is useful if you want to generate a "sitemap" from
    the toctree.
 
+   You can use the ``reverse`` flag option to reverse the order of the entries
+   in the list. This can be useful when using the ``glob`` flag option to
```
Comment (lehmannro) [inline doc/markup/toctree.rst:127]: Minor nit:  The rest of this file uses double spaces after period.

**Comment by lehmannro** [comment]
You can take a look at `tests/test_toctree.py`.  It basically uses the HTML builder to then look at the relations between documents (up/next/previous).  Writing a test for this PR would thus involve cloning `tests/roots/test-glob-toctree` into another testroot, enabling `:reverse:` and checking if the next/previous relationships are now reversed.  You could maybe get away a little simpler with looking at the environment's toctrees (`builder.env.toctree_includes`).

**Comment by tk0miya** [comment]
Ping. any updates?
We will release major version in this autumn (now planning).

It would be nice if this PR will be included the release.

File: sphinx/directives/other.py [hunk 2]
```diff
@@ -46,6 +46,7 @@ class TocTree(Directive):
         'includehidden': directives.flag,
         'numbered': int_or_nothing,
         'titlesonly': directives.flag,
+        'reverse': directives.flag,
```
Comment (tk0miya) [inline sphinx/directives/other.py:49]: Which is better `reverse` or `reversed` ? I feel `reversed` is a little better.

**Comment by tk0miya** [comment]
I added a testcases at https://github.com/tk0miya/sphinx/tree/2527_reversed_toctree .
As a test, I renamed the option to `:reversed:` in my branch.

If agreed, I would like to merge the branch. (if not, I'll pick only the testcodes)
Any comments?

**Comment by shimizukawa** [comment]
+1 for `reversed` to conform to other options like `numbered`.

**Comment by JohnVillalovos** [comment]
`reversed` sounds good to me too. Matches up with Python built-in function :)

Thanks for doing unit tests. I got confused on what to do.

**Comment by tk0miya** [comment]
@JohnVillalovos Thank you for comment. I will merge the change by my side later.

**Comment by tk0miya** [comment]
Merged.
Thank you for contribution!

**Pull request merged**